beam-postgres/Database/Beam/Postgres/Syntax.hs

```diff
@@ -373,6 +373,10 @@ instance IsSql92OrderingSyntax PgOrderingSyntax where
   ascOrdering e = PgOrderingSyntax (fromPgExpression e <> emit " ASC")
   descOrdering e = PgOrderingSyntax (fromPgExpression e <> emit " DESC")
 
+instance IsSql2003OrderingSyntax PgOrderingSyntax where
```

Comment:
One thing I worry about here is what happens if you do `nullsFirst (nullsLast asc)`. I think that would generate incorrect sql. I think we could fix this by modifying `PgOrderingSyntax` to be a data type instead of a newtype.

So something like

```haskell
data PgOrderingSyntax
  = PgOrderingSyntax
  { pgOrderingSyntax :: PgSyntax
  , pgOrderingNullOrdering :: Maybe PgNullOrdering
  }
```

And then define

```haskell
fromPgOrdering :: PgOrderingSyntax -> PgSyntax
fromPgOrdering (PgOrderingSyntax s Nothing) = s
fromPgOrdering (PgOrderingSyntax s (Just PgNullsFirst)) = s <> emit " NULLS FIRST"
```

Or something like that? I just want to make sure something sensible happens.